com.unity.render-pipelines.universal/Runtime/DeferredLights.cs

```diff
@@ -382,49 +377,68 @@ void SetupAdditionalLightConstants(CommandBuffer cmd, ref RenderingData renderin
 
         void SetupMatrixConstants(CommandBuffer cmd, ref RenderingData renderingData)
         {
-            Matrix4x4 proj;
-            Matrix4x4 view;
-
-            proj = renderingData.cameraData.camera.projectionMatrix;
-            view = renderingData.cameraData.camera.worldToCameraMatrix;
+            ref CameraData cameraData = ref renderingData.cameraData;
 
             // When reading back from depth texture, we need to scale back from [0; 1] to [-1; 1] as Unity defaults to for GL clip-space depth convention.
             // As well, non-GL platforms render upside-down, we don't need to y-reverse again on GL platforms.
             bool isGL = SystemInfo.graphicsDeviceType == GraphicsDeviceType.OpenGLCore
                      || SystemInfo.graphicsDeviceType == GraphicsDeviceType.OpenGLES2
                      || SystemInfo.graphicsDeviceType == GraphicsDeviceType.OpenGLES3;
 
-            //Matrix4x4 gpuProj = GL.GetGPUProjectionMatrix(proj, false); // This function not work for orthographic projection, so make we our own!
-            Matrix4x4 gpuProj = new Matrix4x4(
-                new Vector4(1.0f, 0.0f, 0.0f, 0.0f),
-                new Vector4(0.0f, (SystemInfo.graphicsUVStartsAtTop || isGL ? 1.0f : -1.0f), 0.0f, 0.0f),
-                new Vector4(0.0f, 0.0f, (SystemInfo.usesReversedZBuffer ? -1.0f : 1.0f) * 0.5f, 0.0f),
-                new Vector4(0.0f, 0.0f, 0.5f, 1.0f)
-            ) * proj;
-
-
-            // xy coordinates in range [-1; 1] go to pixel coordinates.
-            Matrix4x4 toScreen = new Matrix4x4(
-                new Vector4(0.5f * m_RenderWidth, 0.0f, 0.0f, 0.0f),
-                new Vector4(0.0f, 0.5f * m_RenderHeight, 0.0f, 0.0f),
-                new Vector4(0.0f, 0.0f, 1.0f, 0.0f),
-                new Vector4(0.5f * m_RenderWidth, 0.5f * m_RenderHeight, 0.0f, 1.0f)
-            );
+#if ENABLE_VR && ENABLE_XR_MODULE
+            int eyeCount = cameraData.xr.enabled && cameraData.xr.singlePassEnabled ? 2 : 1;
+#else
+            int eyeCount = 1;
+#endif
+            Matrix4x4[] screenToWorld = new Matrix4x4[2];
+            Matrix4x4 proj;
+            Matrix4x4 view;
 
-            Matrix4x4 clipToWorld = Matrix4x4.Inverse(toScreen * gpuProj * view);
-            cmd.SetGlobalMatrix(ShaderConstants._ScreenToWorld, clipToWorld);
+            for (int eyeIndex = 0; eyeIndex < eyeCount; eyeIndex++)
+            {
+#if ENABLE_VR && ENABLE_XR_MODULE
+                if (cameraData.xr.enabled)
+                {
+                    proj = cameraData.xr.GetProjMatrix(eyeIndex);
```

Comment:
I would suggest to use GetProjectionMatrix instead:
https://github.com/Unity-Technologies/Graphics/blob/647e4729c23dfb911cbf6120b76273aaa87dd8e2/com.unity.render-pipelines.universal/Runtime/UniversalRenderPipelineCore.cs#L75